mobile/lib/blocs/feedback/feedback_bloc.dart

```diff
@@ -1,221 +1,176 @@
 import 'package:app/models/models.dart';
+import 'package:app/services/services.dart';
+import 'package:app/utils/utils.dart';
 import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 part 'feedback_event.dart';
 part 'feedback_state.dart';
 
 class FeedbackBloc extends Bloc<FeedbackEvent, FeedbackState> {
-  FeedbackBloc()
-      : super(
-          const FeedbackState(
-            feedbackType: FeedbackType.none,
-            feedbackChannel: FeedbackChannel.none,
-            contact: '',
-            feedback: '',
-            loading: false,
-          ),
-        ) {
+  FeedbackBloc() : super(const FeedbackState.initial()) {
+    on<InitializeFeedback>(_onInitializeFeedback);
     on<SetFeedbackType>(_onSetFeedbackType);
-    on<SetFeedbackContact>(_onSetFeedbackContact);
-    on<GoToTypeStep>(_onGoToTypeStep);
     on<GoToChannelStep>(_onGoToChannelStep);
     on<GoToFormStep>(_onGoToFormStep);
-    on<SetFeedbackChannel>(_onSetFeedbackChannel);
-    on<FeedbackFormError>(_onFeedbackFormError);
+    on<SubmitFeedback>(_onSubmitFeedback);
+    on<GoToTypeStep>(_onGoToTypeStep);
+    on<SetFeedbackContact>(_onSetFeedbackContact);
     on<SetFeedback>(_onSetFeedback);
-    on<ClearFeedback>(_onClearFeedback);
+    on<SetFeedbackChannel>(_onSetFeedbackChannel);
   }
 
-  void _onFeedbackFormError(
-    FeedbackFormError event,
+  Future<void> _onInitializeFeedback(
+    InitializeFeedback _,
     Emitter<FeedbackState> emit,
-  ) {
-    emit(
-      FeedbackLoadingState(
-        feedbackType: state.feedbackType,
-        feedbackChannel: state.feedbackChannel,
-        contact: state.contact,
-        feedback: state.feedback,
-        loading: true,
-      ),
-    );
+  ) async {
+    final profile = await Profile.getProfile();
+
+    if (profile.emailAddress.isEmpty) {
+      return;
+    }
+
+    if (state.blocStatus == BlocStatus.success) {
+      return emit(
+        FeedbackState.initial().copyWith(
+          emailAddress: profile.emailAddress,
+          feedbackChannel: FeedbackChannel.email,
+        ),
+      );
+    }
```

Comment:
Setting default value for email address incase the user signed in using their email address.